velox/exec/Exchange.cpp

```diff
@@ -14,17 +14,29 @@
  * limitations under the License.
  */
 #include "velox/exec/Exchange.h"
+#include <algorithm>
+#include <random>
 #include "velox/exec/Task.h"
 
 namespace facebook::velox::exec {
 
+void Exchange::addTaskIds(std::vector<std::string>& taskIds) {
+  std::shuffle(
+      std::begin(taskIds), std::end(taskIds), std::default_random_engine{});
```

Comment:
This is not random.  Keep `default_random_engine` as a field and initialize it with `std::random_device()`

Reply:
+1